pkg/node/address.go

```diff
@@ -423,6 +425,12 @@ func AutoComplete() error {
 
 	InitDefaultPrefix(option.Config.DirectRoutingDevice)
```

Comment:
@mvisonneau  I'm a bit lost about how this work, could you share more on that? I assume the cidr is generated in `InitDefaultPrefix`, but it's not skipped.
https://github.com/cilium/cilium/pull/23048/files#diff-9143180298a1f5ba5dc10fd98d87945d1381ef45b7347a639f85e13e84285ca2L424

Reply:
absolutely, I did not find it useful to skip the `InitDefaultPrefix` function call during my tests. I assumed that the allocation was happening later ([L633](https://github.com/cilium/cilium/pull/23048/files#diff-9143180298a1f5ba5dc10fd98d87945d1381ef45b7347a639f85e13e84285ca2R633) &/or [L644](https://github.com/cilium/cilium/pull/23048/files#diff-9143180298a1f5ba5dc10fd98d87945d1381ef45b7347a639f85e13e84285ca2R644)).

As we return early ([L431](https://github.com/cilium/cilium/pull/23048/files#diff-9143180298a1f5ba5dc10fd98d87945d1381ef45b7347a639f85e13e84285ca2R431)) it seems to be sufficient. Not sure about the impact of moving the return logic before the  `InitDefaultPrefix` call but happy to try it out if you want/prefer.

Reply:
Thanks for clarifying.
Looks like there are two places calling `InitDefaultPrefix`:
https://github.com/cilium/cilium/blob/5ff3d72c8b52be779d7bfab8b6e6abfc9f3b8137/pkg/node/address.go#L426  
https://github.com/cilium/cilium/blob/5ff3d72c8b52be779d7bfab8b6e6abfc9f3b8137/daemon/cmd/daemon_main.go#L1404  
I tried the patch in Alibabacloud mode and found that I have to skip both `InitDefaultPrefix` calls to omit the podCIDR. Not sure if there are some differences between ENI mode.